addon/ng2/utilities/get-dependent-files.ts

```diff
@@ -0,0 +1,109 @@
+'use strict';
+
+import * as fs from 'fs';
+import * as path from 'path';
+import * as ts from 'typescript';
+
+/**
+ * Declare a MapType where key is a string and value is an array of type string
+ * Use for storing a file's path as key and its dependent module's relative path
+ */
+declare type ImportMap {
+  [key: string]: ImportMap | string[];
+}
+
+/**
+ * Create a SourceFile as defined by Typescript Compiler API.
+ * Generate a AST structure from a source file.
+ * 
+ * @param fileName source file for which AST is to be extracted
+ */
+export function createtsSourceFile(fileName: string): ts.SourceFile {
+  return ts.createSourceFile(fileName,
+                             fs.readFileSync(fileName).toString(),
+                             ts.ScriptTarget.ES6,
+                             true);
+}
+
+/**
+ *  Store all the files' path under the given path in a passed in array.
+ *  All the files in the subdirectories of the given path are also included.
+ * 
+ * @param filePath path from where all the files' path are retrieved
+ * @param allFiles array where the files are stored and then returned.
+ * 
+ */
+
+export function getFiles(filePath: string, allFiles: string[]): any {
```

Comment:
This look hugely like a broken implementation of the `glob` node module. Check it out.  This implementation seems broken because you automatically assume anything without an extension is a directory, which is wrong (directories can have extensions and files can be without extensions).

In any case, I recommend removing this and using `glob`, which is already installed. This function can be simplified to ```typescript
const allFiles = glob.sync(path.join(somePath, '**/*'), { nodir: true });
```.